modules/transport/Transport.js

```diff
@@ -0,0 +1,191 @@
+import EventEmitter from 'events';
+
+import {
+    MESSAGE_TYPE_EVENT,
+    MESSAGE_TYPE_RESPONSE,
+    MESSAGE_TYPE_REQUEST
+} from './constants';
+import PostMessageTransport from './PostMessageTransport';
+
+/**
+ * Stores the currnet transport that have to be used.
+ */
+export default class Transport extends EventEmitter {
+    /**
+     * Creates new instance.
+     *
+     * @param {Object} options - Optional parameters for configuration of the
+     * transport.
+     */
+    constructor(options = {}) {
+        super();
+
+        const { defaultTransportOptions, noDefaultTransport, transport }
+            = options;
+
+        // We need this because addListener in overriden
+        this.on = this.addListener;
+
+        this._requestID = 0;
```

Comment:
Yes. Good catch! 

My idea was to increase this on every request but obviously I forgot this :) I'll fix it!